### PR TITLE
feat(gigya/globalEvents): move class to function and remove componentWillMount usage

### PR DESCRIPTION
**Pull request by midudev**
Use a function instead a class in order to remove the componentWillMount life cycle and simplify the component.

### CODE REVIEW

File: components/gigya/globalEvents/src/index.js [hunk 1]
```diff
@@ -8,34 +8,24 @@ import {
...
     onLogout && removeOnLogoutSubscriber(onLogout)
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line
-    this._unregisterEvents(this.props)
-    this._registerEvents(nextProps)
-  }
-
-  UNSAFE_componentWillMount() { // eslint-disable-line
+  useEffect(function() {
```
Comment (giodelabarrera) [inline components/gigya/globalEvents/src/index.js]: He visto que haz hecho esto en otros useEffects, el usar una declaración de función y una expresión con arrow function. Es por alguna ventaja o gusto personal? 🙂

File: components/gigya/globalEvents/src/index.js [hunk 2]
```diff
@@ -8,34 +8,24 @@ import {
...
     initListener()
-    this._registerEvents(this.props)
-  }
-
-  componentWillUnmount() {
-    this._unregisterEvents(this.props)
-  }
+    registerEvents({onLogin, onLogout})
```
Comment (giodelabarrera) [inline components/gigya/globalEvents/src/index.js]: Maybe you can move the functions `registerEvents` and `unregisterEvents` inside the useEffect. 
Didn't you skip the linter or have you wanted to ignore the deps of useEffect in an intentional way?

File: components/gigya/globalEvents/src/index.js (hunk 1 above)
Comment (midudev) [reply components/gigya/globalEvents/src/index.js]: Gusto personal. :D

File: components/gigya/globalEvents/src/index.js (hunk 2 above)
Comment (midudev) [reply components/gigya/globalEvents/src/index.js]: I'm just keeping the functionality of before that was using the `componentWillReceiveProps` without any check in order to register again the events.

**Pull request merged**